dotnet/src/Agents/OpenAI/OpenAIResponseAgent.cs

```diff
@@ -144,5 +144,50 @@ private async IAsyncEnumerable<ChatMessageContent> InternalInvokeAsync(
             yield return message;
         }
     }
+
+    // TODO Share this implementation with CoreKernelArgumentsExtensions 
+    private static readonly Dictionary<string, PromptExecutionSettings> s_emptySettings = [];
+
+    /// <summary>
+    /// Provides a merged instance of <see cref="KernelArguments"/> with precedence for override arguments.
+    /// </summary>
+    /// <param name="overrideArguments">The override arguments.</param>
+    /// <remarks>
+    /// This merge preserves original <see cref="PromptExecutionSettings"/> and <see cref="KernelArguments"/> parameters.
+    /// It allows for incremental addition or replacement of specific parameters while also preserving the ability
+    /// to override the execution settings.
+    /// </remarks>
+    private KernelArguments MergeArguments(KernelArguments? overrideArguments)
```

Comment:
Suggest adding UT to ensure this behavior.

```suggestion
    private KernelArguments MergeArguments(KernelArguments? overrideArguments)
```